Fix pitch bend buttons and rate/volume knob on Hercules DJ Console Mk1

SamWhited opened the pull request:
Previously the pitch bend buttons were mapped the same way as the jog wheels, however this is identical to the jog wheels right below them and felt a bit useless. This patch maps the buttons to their original purpose, bumping the pitch up or down, and adds a setting that allows reverting to the previous behavior for backwards compatibility.

I only have one of these temporarily and won't be able to do long term maintenance on this controller, but if there are any other changes that need to be made or tested I have one for the next day or two.

SamWhited committed:
Fix pitch bend buttons on Hercules DJ Console Mk1

Previously the pitch bend buttons were mapped the same way as the jog
wheels, however this is identical to the jog wheels right below them and
felt a bit useless. This patch maps the buttons back to bumping the
pitch up or down, and adds a setting that allows reverting to the
previous behavior for backwards compatibility.

Signed-off-by: Sam Whited <sam@samwhited.com>

Swiftb0y commented on res/controllers/Hercules-DJ-Console-Mk1-hid-scripts.js:
```diff
@@ -84,20 +84,28 @@ HerculesMk1Hid.init = function() {
+                engine.setValue(g, "jog", -3);
+            }
+        } else {
+            script.toggleControl(g, "pitch_down");
```
using `toggleControl` has the problem that in case a message gets lost, the assumed state of the controller and the device would become disconnected. Please do `engine.setParameter(g, "pitch_down", v > 0);` instead.

SamWhited replied:
Oh right, that makes perfect sense, I was forgetting that this would obviously hit the "on" and "off" states as well and probably happen twice. Done!

Swiftb0y commented on res/controllers/Hercules DJ Console Mk1.hid.xml:
```diff
@@ -7,6 +7,23 @@
         <forums>http://www.mixxx.org/forums/viewtopic.php?f=7&amp;t=4081</forums>
         <manual>hercules_dj_console_mk1</manual>
     </info>
+    <settings>
+        <group label="Backwards Compatibility">
+            <row orientation="vertical">
+                <option
+                    variable="pitchBendAsJogWheel"
```
please also document this setting (as well as the new behavior in 2.5.1) in the manual.

SamWhited replied:
Currently there is no manual page for this controller. I may or may not be able to find the time to make one since that's a much bigger change, but I'd like to do a few patches first since I only have this controller for ~12 more hours while I do a repair on it.

Swiftb0y reviewed (approved):
LGTM. Thank you.

Swiftb0y commented:
Since you already have the controller, does it still work as described on 2.5? The last time someone changed this mapping and actually had access to the hardware was likely when the mapping got added in 2013...

SamWhited commented:
I have a few more small patches like this that I'd like to make; would it be better to do them all at once in this PR, or make separate small PRs for ease of review?

Thanks for taking a look at this so quickly, I only have the controller for a short amount of time, so this was great timing!

Swiftb0y commented:
Nah, just do it in one PR and split into appropriate commits. Consider using fixup commits for incorporating review changes.

SamWhited commented:
> Since you already have the controller, does it still work as described on 2.5? The last time someone changed this mapping and actually had access to the hardware was likely when the mapping got added in 2013...

As far as I can tell everything works as expected except the LEDs (which as far as I can tell from the discussion and the mapping have never worked correctly). However, there are a few quirks of the mapping like the one fixed in this PR.

SamWhited committed:
Change volume encoder from rate to pregain

This updates the encoder labeled "volume" to tweak the pregain instead
of the tempo. While neither of these is technically "volume", pre-gain
is normally a rotary encoder and volume is normally a linear encoder.
Mapping the knob this way will meet user expectations.

Signed-off-by: Sam Whited <sam@samwhited.com>

SamWhited commented:
> Nah, just do it in one PR and split into appropriate commits

In that case, I've added another commit swapping the "volume" rotary encoder, which was previously mapped to the tempo, to be the gain instead. I know gain *technically* also isn't volume, which is how the control is labeled, but I suspect that's what it was originally used for since actual volume is normally a linear encoder, and I think this will best meet user expectations (also actual volume is currently on the slider, so no need for having two controls mapped to it).

SamWhited commented:
I think I'm done for now actually; there's a ton more that could be done to this controller (the "1", "2", and "3" buttons across the top don't make a lot of sense IMO, there are a few buttons in the middle that are mappable but appear unused, etc.), but this at least makes it somewhat usable and I don't know if it's really worth it to go to deep on a controller that most people probably aren't using.

Swiftb0y reviewed (approved):
honestly. LGTM for now. Anything else you wanna do to it?

SamWhited commented:
> Anything else you wanna do to it?

I may come back to it later if I get bored, but for now I think this at least makes it somewhat usable again and gets rid of the duplicate controls; good to merge.

SamWhited committed:
Explicitly pass a number instead of a boolean

Co-authored-by: Swiftb0y <12380386+Swiftb0y@users.noreply.github.com>

SamWhited commented on res/controllers/Hercules-DJ-Console-Mk1-hid-scripts.js:
```diff
@@ -84,20 +84,26 @@ HerculesMk1Hid.init = function() {
+                engine.setValue(g, "jog", -3);
+            }
+        } else {
+            engine.setParameter(g, "pitch_down", v > 0);
```
Done! Assuming they actually follow the ECMAScript standard this should work either way, but also this is one of the reasons I deeply dislike JavaScript and all its variants and derivatives, so being explicit always sounds good t me.

Swiftb0y reviewed (approved):
Thank you.

SamWhited commented:
Anything left on this one while we're merging stuff? It has an approval and all checks are green. Thanks!

Swiftb0y commented:
Nope. Was just waiting to give others a chance to have a look in case they'd like to. Thanks for the ping.
